pkg/api/v1alpha1/cluster.go

```diff
@@ -833,3 +834,11 @@ func validateMDUpgradeRolloutStrategy(w *WorkerNodeGroupConfiguration) error {
 
 	return nil
 }
+
+// IsRollingNotScale compares current CAPI objects and desired CAPI objects
+// returns true on rolling upgrade, false on scale upgrade.
+// should error if rolling and scale at the same time
+// todo: IsRollingNotScale
+func IsRollingNotScale(ctx context.Context, client client.Client, cluster *Cluster) (bool, error) {
```

Comment:
Why are we importing the client object here? Also, how do we check for the previous cluster fields? This is the api package so it's more for operations that don't require communicating with the Kubernetes cluster.

Reply:
Raymond moved this method under tinkerbell/upgrade, which should be a better place, since it's very provider specific.